src/figuresClasses.ts

```diff
@@ -1,19 +1,79 @@
 export interface Figure {
-
+  shape: 'triangle' | 'circle' | 'rectangle',
+  color: 'red' | 'green' | 'blue',
+  getArea(): number,
 }
 
-export class Triangle {
+export function checkTriangle(sides:number[]):boolean {
+  const maxSide = Math.max(...sides);
+  let sum = 0;
+
+  sides.forEach((el:number) => {
+    if (el !== maxSide) {
+      sum += el;
+    }
+  });
 
+  return sum >= maxSide;
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  constructor(
+    public shape: 'triangle',
+    public color: 'red' | 'green' | 'blue',
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0 || checkTriangle([a, b, c])) {
+      throw new Error('Incorrect data');
```

Comment:
better to specify the error. you can make a common function to check if all parameters are positive values(because it should be checked in all classes) and separate specific checks